src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2670,5 +2670,225 @@ private interface INegator<T> where T : struct
+            where T : IComparable<T>
+        {
+            for (int i = 0; i < length; i++)
+            {
+                ref T current = ref Unsafe.Add(ref searchSpace, i);
+                if ((lowInclusive.CompareTo(current) <= 0) && (highInclusive.CompareTo(current) >= 0))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        internal static int IndexOfAnyExceptInRange<T>(ref T searchSpace, T lowInclusive, T highInclusive, int length)
+            where T : IComparable<T>
+        {
+            for (int i = 0; i < length; i++)
+            {
+                ref T current = ref Unsafe.Add(ref searchSpace, i);
+                if ((lowInclusive.CompareTo(current) > 0) || (highInclusive.CompareTo(current) < 0))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        internal static int IndexOfAnyInRangeUnsignedNumber<T>(ref T searchSpace, T lowInclusive, T highInclusive, int length)
+            where T : struct, IUnsignedNumber<T>, IComparisonOperators<T, T, bool> =>
+            IndexOfAnyInRangeUnsignedNumber<T, DontNegate<T>>(ref searchSpace, lowInclusive, highInclusive, length);
+
+        internal static int IndexOfAnyExceptInRangeUnsignedNumber<T>(ref T searchSpace, T lowInclusive, T highInclusive, int length)
+            where T : struct, IUnsignedNumber<T>, IComparisonOperators<T, T, bool> =>
+            IndexOfAnyInRangeUnsignedNumber<T, Negate<T>>(ref searchSpace, lowInclusive, highInclusive, length);
+
+        private static int IndexOfAnyInRangeUnsignedNumber<T, TNegator>(ref T searchSpace, T lowInclusive, T highInclusive, int length)
+            where T : struct, IUnsignedNumber<T>, IComparisonOperators<T, T, bool>
+            where TNegator : struct, INegator<T>
+        {
+            // T must be a type whose comparison operator semantics match that of Vector128/256.
+
+            if (!Vector128.IsHardwareAccelerated || length < Vector128<T>.Count)
+            {
+                T rangeInclusive = highInclusive - lowInclusive;
+                for (int i = 0; i < length; i++)
+                {
+                    ref T current = ref Unsafe.Add(ref searchSpace, i);
+                    if (TNegator.NegateIfNeeded((current - lowInclusive) <= rangeInclusive))
+                    {
+                        return i;
+                    }
+                }
+            }
+            else if (!Vector256.IsHardwareAccelerated || length < Vector256<T>.Count)
+            {
+                Vector128<T> inRangeVector, lowVector = Vector128.Create(lowInclusive), rangeVector = Vector128.Create(highInclusive - lowInclusive);
+                ref T current = ref searchSpace;
+                ref T oneVectorAwayFromEnd = ref Unsafe.Add(ref searchSpace, length - Vector128<T>.Count);
```

Comment:
```suggestion
                ref T oneVectorAwayFromEnd = ref Unsafe.Add(ref searchSpace, (uint)(length - Vector128<T>.Count));
```
Tiny little bit more perf 😉 -- avoids the sign-extending move, and just uses a plain mov instruction.

The same on other places (I didn't comment there).